### ISSUE TITLE
`jsPlugins` rules not loaded by LSP server

### ISSUE THREAD

**Issue by hossam-magdy** [oxc-project/oxc#21398]
Extracted from https://github.com/oxc-project/oxc/issues/21385#issuecomment-4237902609, possibly same root cause — the LSP server doesn't correctly resolve relative paths in nested `.oxlintrc.json`.

### Reproduction

`.oxlintrc.json` (in `packages/my-package/`):

```json
{
  "jsPlugins": [
    "eslint-plugin-no-relative-import-paths"
  ],
  "rules": {
    "no-relative-import-paths/no-relative-import-paths": [
      "error",
      { "allowSameFolder": true, "allowedDepth": 1, "rootDir": "src", "prefix": "@" }
    ]
  }
}
```

### Steps

1. Open the monorepo root in VS Code / Cursor
2. Open a file in `packages/my-package/src/` containing `import { Foo } from '../../core/types'`
3. Save with `"source.fixAll.oxc": "always"`
4. Observe: no autofix applied — the relative import stays

### Expected

The `no-relative-import-paths` rule should fire and autofix to `import { Foo } from '@/core/types'`.

### CLI works correctly

```sh
cd packages/my-package
npx oxlint src/ -c .oxlintrc.json --fix
# ✅ Fixes the import
```

### Analysis

The LSP server either doesn't load `jsPlugins` at all when resolving config from a nested directory, or resolves the plugin paths relative to the workspace root instead of the config file's directory. Native Oxlint rules (non-jsPlugin) may have the same issue but are harder to observe since they don't depend on external module resolution.

**Comment by khuongtp** [comment]
This might be unrelated, but is there a way to enable all rules from a jsPlugin and set them to “warn”? Right now I have to configure each rule manually

**Comment by camc314** [comment]
unrelated @khuongtp please make a new issue

**Comment by Sysix** [comment]
`no-relative-import-paths/no-relative-import-paths` is using the `context.getCwd()` method, so your `cd packages/my-package` command changes this value.
Without this command, the `cwd` should match the LSP `cwd`, when opening the monorepo root.

When you are only running `npx oxlint packages/my-package/src/ --fix`, does it also not work?

To avoid the problem for the moment, try to use [Multi-Root Workspaces](https://code.visualstudio.com/docs/editing/workspaces/multi-root-workspaces).
